core/src/main/scala/com/nvidia/spark/rapids/tool/udf/EstimateEventRapidsUDF.scala

```diff
@@ -0,0 +1,148 @@
+
+@Description(name = "estimate_event_rapids",
+  value = "_FUNC_(output_path, event_path) ",
+  extended = "Example:\n  > SELECT estimate_event_rapids(hdfs://..., app_1, hdfs://...)")
+class EstimateEventRapidsUDF extends UDF with Logging {
+  def evaluate(outputDir: String, applicationId: String, eventDir: String,
+      scoreFile: String = "onprem"): Int = {
+    val (exitCode, _, _) =
+      EstimateEventRapidsUDF.estimateLog(outputDir, applicationId, eventDir, scoreFile)
+    exitCode
+  }
+}
+
+object EstimateEventRapidsUDF extends Logging {
+  def estimateLog(outputDir: String, applicationId: String, eventDir: String,
+      scoreFile: String = "onprem",
+      enabledML: Boolean = false): (Int, Seq[QualificationSummaryInfo], String) = {
+    val eventPath = eventDir + "/" + applicationId + "_1"
+    val outputDirectory = outputDir + "/" + applicationId
+    val numOutputRows = 1000
+    val hadoopConf = RapidsToolsConfUtil.newHadoopConf()
+    // timeout: 20min
+    val timeout = Option(1200L)
+    val nThreads = 1
+    val order = "desc"
+
+    val platform = try {
+      val clusterPropsOpt = loadClusterProps("./worker_info.yaml")
+      PlatformFactory.createInstance("", clusterPropsOpt)
+    } catch {
+      case NonFatal(e) =>
+        logError("Error creating the platform", e)
+        return (1, Seq[QualificationSummaryInfo](), "")
+    }
+    val pluginTypeChecker = try {
+      new PluginTypeChecker(
+        platform,
+        None)
+    } catch {
+      case ie: IllegalStateException =>
+        logError("Error creating the plugin type checker!", ie)
+        return (1, Seq[QualificationSummaryInfo](), "")
+    }
+
+    val (eventLogFsFiltered, _) = EventLogPathProcessor.processAllPaths(
+      None, None, List(eventPath), hadoopConf)
+    val filteredLogs = eventLogFsFiltered
+
+    // uiEnabled = false
+    val qual = new Qualification(
+      outputPath=outputDirectory,
+      numRows=numOutputRows,
+      hadoopConf=hadoopConf,
+      timeout= timeout,
+      nThreads=nThreads,
+      order=order,
+      pluginTypeChecker=pluginTypeChecker,
+      reportReadSchema=false,
+      printStdout=false,
+      enablePB=true,
+      reportSqlLevel=false,
+      maxSQLDescLength=100,
+      mlOpsEnabled=false,
+      penalizeTransitions=true,
+      tunerContext=None,
+      clusterReport = false
+    )
+    try {
+      val res: Seq[QualificationSummaryInfo] = qual.qualifyApps(filteredLogs)
+      val predictScore = if (enabledML && res.nonEmpty) {
+        // 传python
+        // outputDirectory
+        execMLPredict(applicationId, outputDirectory)
+      } else ""
+      (0, res, predictScore)
+    } catch {
+      case NonFatal(e) =>
+        logError(s"Error when analyze ${applicationId}, path is ${eventPath}.", e)
+        (1, Seq[QualificationSummaryInfo](), "")
+    }
+  }
+
+  private def execMLPredict(applicationId: String, outputDirectory: String): String = {
```

Comment:
The architecture of the tools is that jar is being executed by the python wrapper.
Putting an exec to run the predict command from within the jar implies that we have both the jar and the wrapper python are calling each other which is inconsistent and duplicating the work.